### PR TITLE
Implement auth cell

### LINKED ISSUE #3: Users should be able to register and authenticate

### PR DESCRIPTION
**Pull request by larribas**
Closes #3 

After we agree on the basics of this, we can probably do a few tasks concurrently. I've opened issues #13, #14 and #15 to cover the whole functionality.

### CODE REVIEW

File: lib/real_world/auth/domain_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        it { is_expected.to be(true) }
+      end
+
+      context 'when the password does not match' do
+        let(:entered_password) { 'invalid!' }
+        it { is_expected.to be(false) }
+      end
+    end
+  end
+end
```
Comment (hecrj) [inline lib/real_world/auth/domain_spec.rb:52]: I love the idea of having unit test files next to the actual code. Go and Rust do it like that and I really think it keeps things easier to understand and maintain!

File: lib/colmena/cell.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        name_without_namespace.gsub(/([^\^])([A-Z])/,'\1_\2').downcase.to_sym
+      end
+    end
+
+    def self.included(klass)
+      klass.extend ClassMethods
+    end
+
+    def initialize(adapters={})
+      @commands = inject_ports(self.class.commands)
```
Comment (hecrj) [inline lib/colmena/cell.rb:39]: We need to initialize the `queries` here too!

```suggestion
      @commands = inject_ports(self.class.commands)
      @queries = inject_ports(self.class.queries)
```

**Comment by hecrj** [comment]
Looks good! :+1:

I will branch from this and do some work on the HTTP layer!

File: lib/colmena/cell.rb (hunk 2 above)
Comment (hecrj) [reply lib/colmena/cell.rb:39]: I already did this in #17.

Comment (larribas) [reply lib/colmena/cell.rb:39]: gr8, thank you!

File: lib/real_world/auth/domain_spec.rb (hunk 1 above)
Comment (larribas) [reply lib/real_world/auth/domain_spec.rb:52]: I know right?

**Pull request merged**